BepInEx.Core/Contract/IPluginLoader.cs

```diff
@@ -0,0 +1,26 @@
+﻿namespace BepInEx;
+
+/// <summary>
+///     An interface that allows to dynamically load and track an assembly
+///     which can be obtained from a provided
+/// </summary>
+public interface IPluginLoader
```

Comment:
Maybe a "Source" property would be useful to track where the plugin came from?

Reply:
What would it contain? The PluginInfo of the provider?

Reply:
Patcher's PluginInfo I guess? I could see it being useful in some cases, mostly debugging.

Reply:
Done, but I had to abstract all base plugins classes to an interface which strangely enough, said interface was already done, but its file was excluded from the solution.

Now, all 3 base plugins classes implements IPlugin which provides the stuff everyone was providing already, but with more consistency. The plugins still obtains their basic attribute metadata, but the rest is offloaded to the chainloader so the plugin can ask for their full info from their guid which will include the Source so all plugins can have all the info regardless of runtime including Source.